sklearn/linear_model/logistic.py

```diff
@@ -608,10 +610,10 @@ def logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
     # and check length
     # Otherwise set them to 1 for all examples
     if sample_weight is not None:
-        sample_weight = np.array(sample_weight, dtype=np.float64, order='C')
+        sample_weight = np.array(sample_weight, dtype=X.dtype, order='C')
```

Comment:
we were discussing with @glemaitre (and @GaelVaroquaux) to force `X.dtype` and `y.dtype` to be the same.